models/models_test.go

```diff
@@ -47,318 +48,322 @@ func (_f MockUtilsIam) DeleteCertificate(certName string) error {
-	fakeiam := iam.New(session)
-	fakeiam.Handlers.Clear()
-	fakeiam.Handlers.Send.PushBack(func(r *request.Request) {
-		//t.Log(r.Operation.Name)
-		switch r.Operation.Name {
-		case "ListServerCertificates":
-			old := time.Now().AddDate(0, 0, -2)
-			current := time.Now().AddDate(0, 0, 0)
-
-			list := []*iam.ServerCertificateMetadata{
-				&iam.ServerCertificateMetadata{
-					Arn: aws.String("an-active-certificate"),
-					ServerCertificateName: aws.String("an-active-certificate"),
-					ServerCertificateId:   aws.String("an-active-certificate"),
-					UploadDate:            &old,
-				},
-				&iam.ServerCertificateMetadata{
-					Arn: aws.String("some-other-active-certificate"),
-					ServerCertificateName: aws.String("some-other-active-certificate"),
-					ServerCertificateId:   aws.String("some-other-active-certificate"),
-					UploadDate:            &old,
-				},
-				&iam.ServerCertificateMetadata{
-					Arn: aws.String("orphaned-but-not-old-enough"),
-					ServerCertificateName: aws.String("orphaned-but-not-old-enough"),
-					ServerCertificateId:   aws.String("this-cert-should-not-be-deleted"),
-					UploadDate:            &current,
-				},
-				&iam.ServerCertificateMetadata{
-					Arn: aws.String("some-orphaned-cert"),
-					ServerCertificateName: aws.String("some-orphaned-cert"),
-					ServerCertificateId:   aws.String("this-cert-should-be-deleted"),
-					UploadDate:            &old,
-				},
-				&iam.ServerCertificateMetadata{
-					Arn: aws.String("some-other-orphaned-cert"),
-					ServerCertificateName: aws.String("some-other-orphaned-cert"),
-					ServerCertificateId:   aws.String("this-cert-should-also-be-deleted"),
-					UploadDate:            &old,
-				},
-			}
-			data := r.Data.(*iam.ListServerCertificatesOutput)
-			data.IsTruncated = aws.Bool(false)
-			data.ServerCertificateMetadataList = list
-		}
-	})
-
-	//mock out the aws call to return a fixed list of distributions
-	fakecf := cloudfront.New(session)
-	fakecf.Handlers.Clear()
-	fakecf.Handlers.Send.PushBack(func(r *request.Request) {
-		//t.Log(r.Operation.Name)
-		switch r.Operation.Name {
-		case "ListDistributions2016_11_25":
-			list := []*cloudfront.DistributionSummary{
-				&cloudfront.DistributionSummary{
-					ARN: aws.String("some-distribution"),
-					ViewerCertificate: &cloudfront.ViewerCertificate{
-						IAMCertificateId: aws.String("an-active-certificate"),
+var _ = Describe("Models", func() {
+	It("should delete orphaned certs", func() {
+		logger := lager.NewLogger("cdn-cron-test")
+		logOutput := bytes.NewBuffer([]byte{})
+		logger.RegisterSink(lager.NewWriterSink(logOutput, lager.ERROR))
+
+		settings, _ := config.NewSettings()
+		session := session.New(nil)
+
+		//mock out the aws call to return a fixed list of certs, two of which should be deleted
+		fakeiam := iam.New(session)
+		fakeiam.Handlers.Clear()
+		fakeiam.Handlers.Send.PushBack(func(r *request.Request) {
+			//t.Log(r.Operation.Name)
```

Comment:
I think it might be worth moving these handler functions out in to a set of fixture files, in order to make the test code itself easier to read and write. We might end up with a small explosion of files, but IMO that's a reasonable tradeoff for being able to quickly comprehend a test.

Reply:
Yes, but out of scope of this PR as it affects actual test changes rather than the test runner/matchers.